src/components/Dropdown/index.js

```diff
@@ -0,0 +1,186 @@
+  items = [],
+  placeholder = 'Select an option',
+  value = null,
+  cssClass = '',
+} = {}) {
+  Component.call(this, { html, events });
+
+  this.placeholder = placeholder;
+  this.items = new Map();
+  this.itemSelected = null;
+
+  this.onSelect = (item) => {
+    const existsAndIsDifferent = this.itemSelected != null && this.itemSelected !== item;
+    if (existsAndIsDifferent) this.itemSelected.unselect();
+
+    this.itemSelected = item;
+    this.setValue(item.getValue());
+    this.emit('select', item);
+
+    if (this.isOpen()) this.close();
+  };
+
+  this.onUnselect = (item) => {
+    const isDifferent = this.itemSelected !== item;
+    if (isDifferent) return;
+
+    this.itemSelected = null;
+    this.setValue(null);
+    this.emit('unselect', item);
+  };
+
+  items.map((item) => this.addItem(item));
+  this.setValue(value);
+
+  this.selected
+    .get('dropdown-toggle')
+    .addEventListener('click', () => this.toggle());
+
+  const closeOnClickOutside = (event) => {
+    const isOutside = !event
+      .composedPath()
+      .includes(this.selected.get('dropdown-container'));
+
+    if (!isOutside) return;
+
+    if (this.isOpen()) this.close();
+  };
+
+  this.listen('mount', () => document.addEventListener('click', closeOnClickOutside),
+  );
+  this.listen('unmount', () => document.removeEventListener('click', closeOnClickOutside),
+  );
+
+  if (cssClass) {
+    this.selected.get('dropdown-container').classList.add(cssClass);
+  }
+}
+
+Dropdown.prototype = Object.assign(Dropdown.prototype, Component.prototype, {
+  addItem(props = {}) {
+    if (this.items.has(props.value)) return;
+
+    const item = new DropdownItem(props);
+    const $list = this.selected.get('dropdown-options');
+    item.mount($list);
+    item.listen('select', this.onSelect);
+    item.listen('unselect', this.onUnselect);
+    this.items.set(props.value, item);
+    this.emit('item:add', item);
+  },
+
+  removeItem(value = '') {
+    if (!this.items.has(value)) return;
+
+    const item = this.items.get(value);
+    item.unmount();
+    this.items.delete(value);
+
+    if (item.getValue() === this.getValue()) {
+      this.setValue(null);
+    }
+
+    this.emit('item:remove', item);
+  },
+
+  isOpen() {
+    return this.selected
+      .get('dropdown-container')
+      .classList.contains('dropdown--open');
+  },
+
+  toggle(condition = this.isOpen()) {
+    if (condition) this.close();
+    else this.open();
+  },
+
+  open() {
+    if (this.items.size === 0) return;
+
+    const $container = this.selected.get('dropdown-container');
+    $container.classList.add('dropdown--open');
+    this.emit('open');
+  },
+
+  close() {
+    const $container = this.selected.get('dropdown-container');
+    $container.classList.remove('dropdown--open');
+    this.emit('close');
+  },
+
+  getValue() {
+    const { value } = this.selected.get('dropdown-selected').dataset;
+    if (value == null) return null;
+    return value;
```

Comment:
??

Porque temos isso? Se vc retorna `null` se for null e no fim vc retorna o `value`, no fim se vc apenas retornar o `value` da na mesma, pq se ele for `null` ele vai retornar `null` anyway

Se isso for pq da comparação `==` sem ser `===` saiba que usar `==` é considerado uma péssima prática